Tests/Fixtures/Controller/ParamsAnnotatedController.php

```diff
@@ -36,12 +39,12 @@ public function getArticlesAction(ParamFetcher $paramFetcher)
     {
     }
 
-    #[QueryParam(name: 'page', requirements: '\d+', default: '1', description: 'Page of the overview')]
+    #[QueryParam(name: 'page', requirements: '\d+', default: '1', description: 'Page of the overview.')]
     #[RequestParam(name: 'byauthor', requirements: '[a-z]+', description: 'by author', incompatibles: ['search'], strict: true)]
-    #[QueryParam(name: 'filters', requirements: '\d+', default: '1', description: 'Page of the overview')]
+    #[QueryParam(name: 'filters', map: true, requirements: new NotNull())]
```

Comment:
based on the CI error I thinnk PHP 8.0 does not support `new NotNull`. This seems to be a PHP 8.1 feature: https://www.php.net/releases/8.1/en.php

Reply:
I'm definitely going for the "just skip on PHP 8.0" option whenever I clean this PR up, otherwise the config between annotations and attributes starts changing and may end up giving different results based on where the config comes from.

Reply:
If it helps I'm ok for simply dropping php 8. 0 support

Reply:
Funny enough, it's just two test cases that need to be skipped because of this.  So, I think we're good on that one for the moment.

Reply:
I'm not sure if possible another possibility may would be create a.

```
#[\Attribute(\Attribute::IS_REPEATABLE | \Attribute::TARGET_CLASS | \Attribute::TARGET_METHOD)]
class QueryParamNotNull extends QueryParam {
    public function __construct(
        string $name = '',
        ?string $key = null,
        $requirements = null, 
        $default = null,
        array $incompatibles = [],
        string $description = '',
        bool $strict = false,
        bool $map = false,
        bool $nullable = false,
        bool $allowBlank = true
    ) {
        parent::__construct(
            $name,
            $key,
            $requirements === null ? new NotNull() : $requirements,
            $default,
            $incompatibles,
            $description,
            $strict,
            $map,
            $nullable,
            $allowBlank 
        );
    }
}
```

Can also just be in the test namespace.